### ISSUE TITLE
how to get cognitoIdentityId to be populated

### ISSUE THREAD

**Issue by raygerrard** [aws-amplify/amplify-js#4925]
** Which Category is your question related to? **
API
** What AWS Services are you utilizing? **
Cognito, Lambda, ApiGateway
** Provide additional details e.g. code snippets **
After adding a userprofile api via the command line, I have set userIdPresent=true.

The cognitoIdentityId is always null even though the user is authenticated. Is there any steps that need to be taken to enable this to be populated?

I have made a POST to the endpoint and checked dynamodb and a record is created but the userID is set to UNAUTH.

```js
if (userIdPresent) {
    req.body['userId'] = req.apiGateway.event.requestContext.identity.cognitoIdentityId || UNAUTH;
} 
```

**Comment by raygerrard** [comment]
I have also added custom request header. I don't want to have to add this to every request but it didn't work when adding this to Amplify.configure.

```js
API.post('profile', '/profile', {
  headers: { Authorization: `Bearer ${(await Auth.currentSession()).getIdToken().getJwtToken()}` }
});
```

I have also added a custom cognito pool Authoriser and added that to ANY request methods and no success.

Can anyone advise on this? I thought this would be an out of the box feature to be able to limit the api to cognito users and be able to get that user inside of lambda.

I have spent a couple of days trying to figure this out and not been able to find a definitive answer.
